app/lib/actions/katello/content_view_version/export.rb

```diff
@@ -0,0 +1,72 @@
+module Actions
+  module Katello
+    module ContentViewVersion
+      class Export < Actions::EntryAction
+        middleware.use Actions::Middleware::KeepCurrentUser
+
+        ISO_OUTPUT_BASEDIR = "/var/lib/pulp/published/yum/master/group_export_distributor/"
+
+        def plan(content_view_version, export_to_iso, since, iso_size)
+          # assemble data to feed to Pulp
+          start_date = since ? since.iso8601 : nil
+          content_view = ::Katello::ContentView.find(content_view_version.content_view_id)
+          org_label = ::Organization.find_by(:id => content_view.organization_id).label
+          group_id = "#{org_label}-#{content_view.label}-"\
+                     "v#{content_view_version.major}.#{content_view_version.minor}"
+
+          repo_pulp_ids = content_view_version.archived_repos.
+                            select { |r| r.content_type == 'yum' }.collect { |r| r.pulp_id }
```

Comment:
Correct, there would be two things to solve for puppet repos:

* adding the "regular" puppet distributor to repos
* allowing ISOs to be created with multiple types of content

The former is being handled as part of the nodes work, but the latter is being tracked in https://pulp.plan.io/issues/1544. It's technically possible to do puppet export/import using only the new distributor but I think it would be a better experience if we waited for 1544.